packages/block-library/src/navigation/block.json

```diff
@@ -9,7 +9,7 @@
 	"textdomain": "default",
 	"attributes": {
 		"ref": {
-			"type": "number"
+			"type": "string"
```

Comment:
@draganescu I've been thinking about this. We can't actually perform a migration on the block to cover this because it requires dynamic data about the Post (i.e. the `slug`). 

It's possible to automatically update Nav blocks to use the slug that comes back from the initial API request as the `ref`. But until that point the data stored in the block will not conform to the attributes schema.

Reply:
Also should we allow blocks created with an ID to continue to use that ID rather than force them to be auto-migrated to use the slug?

Reply:
This is a reply to [a conversation in Slack](https://wordpress.slack.com/archives/C01KDAZJMQ9/p1661943843305499?thread_ts=1661940484.527829&cid=C01KDAZJMQ9)

Reply:
> But until that point the data stored in the block will not conform to the attributes schema.

This is fine as it represents a normal migration: open in editor -> migrate.